models/tiegcm/model_mod.f90

```diff
@@ -396,371 +260,247 @@ function get_model_size()
+      ! We only need the height (aka level)
+      ! the obs_def_upper_atm_mod.f90:get_expected_O_N2_ratio routines queries
+      ! for the pressure at the model levels - EXACTLY - so ...
+      ! FIXME ... at present ... the only time model_interpolate
+      ! gets called with QTY_PRESSURE is to calculate density, which
+      ! requires other variables that only live on the midpoints.
+      ! I cannot figure out how to generically decide when to
+      ! use plevs vs. pilevs
+
+      ! Check to make sure vertical level is possible.
+      if ((level < 1) .or. (level > nlev)) then
+         istatus(:) = 33
+         return
+      else
+         obs_val(:) = plevs(level)
+         istatus(:) = 0
+      endif
+   elseif (which_vert == VERTISHEIGHT) then
 
-if(     vert_is_height(location) ) then
-   height = lon_lat_lev(3)
-elseif( vert_is_level(location) ) then
-   level  = int(lon_lat_lev(3))
-elseif( vert_is_undef(location) ) then
-   height = MISSING_R8
-   level  = -1
-else
-   which_vert = nint(query_location(location))
-   write(string1,*) 'vertical coordinate type:',which_vert,' cannot be handled'
-   call error_handler(E_ERR,'model_interpolate',string1,source,revision,revdate)
-endif
+      ! If PRESSURE; calculate the pressure from several variables. HK I don't understand this.
 
-! Check to make sure vertical level is possible.
-if (vert_is_level(location)) then
-   if ((level < 1) .or. (level > nilev)) return
-endif
 
-if ((ikind == QTY_PRESSURE) .and. (vert_is_level(location))) then
-   ! Some variables need plevs, some need pilevs
-   ! We only need the height (aka level)
-   ! the obs_def_upper_atm_mod.f90:get_expected_O_N2_ratio routines queries
-   ! for the pressure at the model levels - EXACTLY - so ...
-   ! FIXME ... at present ... the only time model_interpolate
-   ! gets called with QTY_PRESSURE is to calculate density, which
-   ! requires other variables that only live on the midpoints.
-   ! I cannot figure out how to generically decide when to
-   ! use plevs vs. pilevs 
-   if (level <= nlev) then
-      obs_val = plevs(level)
-      istatus = 0
-   endif
-   return
-endif
+      ! FIXME ... is it possible to try to get a pressure with which_vert == undefined
+      ! At present, vert_interp will simply fail because height is a negative number.
+      ! HK what are you supposed to do for pressure with VERTISUNDEF? level 1?
 
-! Get lon and lat grid specs
-bot_lon        = lons(1)                         ! 180.
-delta_lon      = abs((lons(1)-lons(2)))          ! 5. or 2.5
-zero_lon_index = int(bot_lon/delta_lon) + 1      ! 37 or 73
-top_lon        = lons(nlon)                      ! 175. or 177.5
-bot_lat        = lats(1)                         !
-top_lat        = lats(nlat)                      !
-delta_lat      = abs((lats(1)-lats(2)))          !
+      ! HK pressure is not in the state so do not call get_state.
+      call error_handler(E_ERR, 'model_interpolate', 'vert is height for pressure interpolate')
```

Comment:
if this is an error message, it needs to say why it's failing.  "pressure obs with vert in height not supported"  or something?

Reply:
I'll strip this out, this is me catching pressure when testing.  Sorry.

Reply:
removed.